src/Microsoft.AspNet.Testing/xunit/OSSkipConditionAttribute.cs

```diff
@@ -2,62 +2,81 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Dnx.Runtime;
 
 namespace Microsoft.AspNet.Testing.xunit
 {
-    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true)]
     public class OSSkipConditionAttribute : Attribute, ITestCondition
     {
-        private readonly OperatingSystems _excludedOS;
+        private readonly OperatingSystems _excludedOperatingSystem;
+        private readonly IEnumerable<string> _excludedVersions;
 
-        public OSSkipConditionAttribute(OperatingSystems excludedOperatingSystems)
+        public OSSkipConditionAttribute(OperatingSystems operatingSystem, params string[] versions)
         {
-            _excludedOS = excludedOperatingSystems;
+            _excludedOperatingSystem = operatingSystem;
+            _excludedVersions = versions ?? Enumerable.Empty<string>();
+            RuntimeEnvironment = TestPlatformHelper.RuntimeEnvironment;
         }
 
+        // to enable unit testing
+        internal IRuntimeEnvironment RuntimeEnvironment { get; set; }
```

Comment:
Make an internal constructor that accepts this in addition to the other two params